pkg/loop/internal/reporting_plugin_service.go

```diff
@@ -26,29 +26,44 @@ func NewReportingPluginServiceClient(broker Broker, brokerCfg BrokerConfig, conn
 	return &ReportingPluginServiceClient{pluginClient: pc, reportingPluginService: pb.NewReportingPluginServiceClient(pc), serviceClient: newServiceClient(pc.brokerExt, pc)}
 }
 
-func (m *ReportingPluginServiceClient) NewReportingPluginFactory(ctx context.Context, config types.ReportingPluginServiceConfig, grpcProvider grpc.ClientConnInterface, pipelineRunner types.PipelineRunnerService, errorLog types.ErrorLog) (types.ReportingPluginFactory, error) {
+func (m *ReportingPluginServiceClient) NewReportingPluginFactory(
+	ctx context.Context,
+	config types.ReportingPluginServiceConfig,
+	grpcProvider grpc.ClientConnInterface,
+	pipelineRunner types.PipelineRunnerService,
+	telemetry types.TelemetryService,
+	errorLog types.ErrorLog,
+) (types.ReportingPluginFactory, error) {
 	cc := m.newClientConn("ReportingPluginServiceFactory", func(ctx context.Context) (id uint32, deps resources, err error) {
 		providerID, providerRes, err := m.serve("PluginProvider", proxy.NewProxy(grpcProvider))
 		if err != nil {
 			return 0, nil, err
 		}
 		deps.Add(providerRes)
 
-		errorLogID, errorLogRes, err := m.serveNew("ErrorLog", func(s *grpc.Server) {
-			pb.RegisterErrorLogServer(s, &errorLogServer{impl: errorLog})
+		pipelineRunnerID, pipelineRunnerRes, err := m.serveNew("PipelineRunner", func(s *grpc.Server) {
```

Comment:
Sorry for the noisy diff here: I reordered the deps in the order in which they appear at the top of the function.